alg/sha256_arm.c

```diff
@@ -98,8 +87,12 @@ SHA256_Transform_arm(uint32_t state[static restrict 8],
 #endif
 {
 	uint32x4_t Y[4];
+	uint32x4_t S0, S1; /* S0 and S1 alternate between ABEF and CDGH. */
+	uint32x4_t S0_orig, S1_orig;
```

Comment:
I would be inclined to make this `uint32x4_t _state[2]`.  I often use `_` prefixes for "this is the same thing as the function parameter but reinterpreted`.

Reply:
done

Reply:
To clarify the history: I rebased this change into the "use intrinsics for round calculation".  The other improvements had to be in their own commits.